jax/_src/lax/linalg.py

```diff
@@ -194,7 +198,7 @@ def lu_pivots_to_permutation(pivots, permutation_size: int):
   return permutation
 
 
-def lu(x):
+def lu(x: ArrayLike) -> Tuple[Array, Array, Array]:
```

Comment:
Why are some functions in `lax` consume `Array` and others consuming `ArrayLike`?

Reply:
Some consume `Array` because they access array attributes (like `shape`) before passing the input to a function that converts it to an array.

Reply:
I chose to annotate the existing implementation here rather than to change the implementation for better consistency.